packages/ember-template-compiler/lib/system/compile-options.ts

```diff
@@ -102,28 +107,29 @@ function wrapLegacyPluginIfNeeded(_plugin: PluginFunc | LegacyPluginClass): Plug
       let pluginInstantiated = false;
 
       return {
-        name: _plugin.constructor && _plugin.constructor.name,
+        name: plugin.name,
```

Comment:
Pretty sure this was a big previously: if `plugin` is a class, then `plugin.constructor` would have been `Function`, so this would always be the string `"Function"`.

Reply:
(Hopefully the name wasn't used to deduplicate plugins!)